pkg/receive/multitsdb.go

```diff
@@ -383,9 +412,23 @@ func (t *MultiTSDB) pruneTSDB(ctx context.Context, logger log.Logger, tenantInst
 	tenantTSDB.mtx.Lock()
 	defer tenantTSDB.mtx.Unlock()
 
-	// Lock the entire tenant to make sure the shipper is not running in parallel.
+	// Make sure the shipper is not running in parallel.
 	tenantInstance.mtx.Lock()
-	defer tenantInstance.mtx.Unlock()
+	shipper := tenantInstance.ship
+	tenantInstance.ship = nil
+	tenantInstance.mtx.Unlock()
+	shipper.DisableWait()
```

Comment:
Maybe this is a bit safer, and it's also done the same way in the defer bellow.

```suggestion
	shipper.DisableWait()
        tenantInstance.mtx.Unlock()
```